pkg/dotgit/remote/untag/untag.go

```diff
@@ -0,0 +1,200 @@
+package untag
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"slices"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+
+	ls "github.com/MaineK00n/vuls-data-update/pkg/dotgit/ls/remote"
+)
+
+type options struct {
+	orasHost   string
+	baseAPIURL string
+}
+
+type Option interface {
+	apply(*options)
+}
+
+type baseAPIURLOption string
+
+func (u baseAPIURLOption) apply(opts *options) {
+	opts.baseAPIURL = string(u)
+}
+
+func WithBaseAPIURL(url string) Option {
+	return baseAPIURLOption(url)
+}
+
+type orasHostOption string
+
+func (u orasHostOption) apply(opts *options) {
+	opts.orasHost = string(u)
+}
+
+func WithOrasHost(host string) Option {
+	return orasHostOption(host)
+}
+
+func Untag(imageRef, token string, opts ...Option) error {
+	options := &options{
+		baseAPIURL: "https://api.github.com",
+		orasHost:   "ghcr.io",
+	}
+
+	for _, o := range opts {
+		o.apply(options)
+	}
+
+	log.Printf("[INFO] Untag dotgit %s", imageRef)
+
+	ctx := context.TODO()
+
+	index := strings.LastIndex(imageRef, ":")
+	if index == -1 {
+		return errors.Errorf("unexpected image format. expected: %q, actual: %q", "ghcr.io/<owner>/<package>:tag", imageRef)
+	}
+
+	repoRef := imageRef[:index]
+	tag := imageRef[index+1:]
```

Comment:
When receiving input like `<repository>:<tag>@<digest>`, is it necessary to remove the `@<digest>` portion?